WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -0,0 +1,318 @@
+///
+final class CollectOrderPaymentUseCase: NSObject {
+
+    /// Store's ID.
+    ///
+    private let siteID: Int64
+
+    /// Order to collect.
+    ///
+    private let order: Order
+
+    /// Formatted amount to collect.
+    ///
+    private let formattedAmount: String
+
+    /// Payment Gateway Account to use.
+    ///
+    private let paymentGatewayAccount: PaymentGatewayAccount
+
+    /// Stores manager.
+    ///
+    private let stores: StoresManager
+
+    /// Analytics manager,
+    ///
+    private let analytics: Analytics
+
+    /// View Controller used to present alerts.
+    ///
+    private var rootViewController: UIViewController
+
+    /// Stores the card reader listener subscription while trying to connect to one.
+    ///
+    private var readerSubscription: AnyCancellable?
+
+    /// Closure to inform when the full flow has been completed, after receipt management.
+    /// Needed to be saved as an instance variable because it needs to be referenced from the `MailComposer` delegate.
+    ///
+    private var onCompleted: (() -> ())?
+
+    /// Alert manager to inform merchants about reader & card actions.
+    ///
+    private lazy var alerts = OrderDetailsPaymentAlerts(presentingController: rootViewController)
+
+    /// IPP payments collector.
+    ///
+    private lazy var paymentOrchestrator = PaymentCaptureOrchestrator()
+
+    /// Controller to connect a card reader.
+    ///
+    private lazy var connectionController = {
+        CardReaderConnectionController(forSiteID: siteID,
+                                       knownReaderProvider: CardReaderSettingsKnownReaderStorage(), alertsProvider: CardReaderSettingsAlerts())
+    }()
+
+    init(siteID: Int64,
+         order: Order,
+         formattedAmount: String,
+         paymentGatewayAccount: PaymentGatewayAccount,
+         rootViewController: UIViewController,
+         stores: StoresManager = ServiceLocator.stores,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.siteID = siteID
+        self.order = order
+        self.formattedAmount = formattedAmount
+        self.paymentGatewayAccount = paymentGatewayAccount
+        self.rootViewController = rootViewController
+        self.stores = stores
+        self.analytics = analytics
+    }
+
+    /// Starts the collect payment flow.
+    /// 1. Connects to a reader
+    /// 2. Collect payment from order
+    /// 3. If successful: prints or emails receipt
+    /// 4. If failure: Allows retry
+    ///
+    ///
+    /// - Parameter onCollect: Closure Invoked after the collect process has finished.
+    /// - Parameter onCompleted: Closure Invoked after the flow has been totally completed, Currently after merchant has handled the receipt.
+    // TODO: Remember to check why the amount is provided in order details view model
+    func collectPayment(onCollect: @escaping (Result<Void, Error>) -> (), onCompleted: @escaping () -> ()) {
```

Comment:
Is onCollect necessary? Looks empty below

Reply:
This is not necessary for Simple Payments, but will be for collecting payments on Order Detail, as we need to update the view right after collecting the payment. 

https://github.com/woocommerce/woocommerce-ios/blob/c3edcbeb1f405a8949f99966a7632f5312e55a65/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/OrderDetailsViewController.swift#L765-L767